### PR TITLE
Create TransformationMapping before Item (FIX CI FAILURE)

### PR DESCRIPTION
**Pull request by lpichler**
FIXing CI failure https://travis-ci.org/ManageIQ/manageiq-api/jobs/578155651

After this https://github.com/ManageIQ/manageiq/pull/19204 it looks like that when we are creating 

`TransformationMapping` with `TransformationMappingItems`( with [DataStorage](https://github.com/ManageIQ/manageiq/blob/50c0ca0bb7a18aa6fea88ff44c579c716ce3fe56/app/models/transformation_mapping_item.rb) which requires created `TransformationMapping `)

we need to create first
`TransformationMapping `

and then create `TransformationMappingItems ` and then assign `TransformationMapping `

It looks like that there is also issue  in create operation https://github.com/ManageIQ/manageiq-api/blob/ba3258bef9b483b3c0cf5d9c950f62d235bbba91/app/controllers/api/transformation_mappings_controller.rb 

please review @agrare @thearifsmail @djberg96 

@miq-bot assign @abellotti

### CODE REVIEW

**Comment by agrare** [comment]
@lpichler this looks like a dup of https://github.com/ManageIQ/manageiq-api/pull/664

**Comment by lpichler** [comment]
@agrare yes, expect to specs which are needed for fix CI failure.

@djberg96 
CI is broken because of 
-  `edit_resource(type, id, data)` method (@djberg96 you have it covered in your implementation)
- improver creation of TransformationMappingItem what I have in this PR https://github.com/ManageIQ/manageiq-api/pull/666/files#diff-7a31d90ba468e1a90cbdb4a2699f5691R13 
 

@djberg96 so can you take my changes about `FactoryBot.create` in order to fix CI in one PR ?

thanks

**Comment by agrare** [comment]
@lpichler I think you need to do the same thing for `create_resource`

File: app/controllers/api/transformation_mappings_controller.rb [hunk 1]
```diff
@@ -13,12 +13,10 @@ def create_resource(_type, _id, data = {})
...
       raise "Must specify transformation_mapping_items" unless data["transformation_mapping_items"]
       transformation_mapping = resource_search(id, type, collection_class(type))
-
       updated_data = data.except("transformation_mapping_items")
       transformation_mapping.update_attributes!(updated_data) if updated_data.present?
-
-      transformation_mapping.transformation_mapping_items = create_mapping_items(data["transformation_mapping_items"])
-      transformation_mapping.save!
+      transformation_mapping_items = create_mapping_items(data["transformation_mapping_items"])
+      transformation_mapping_items.each { |tmi| tmi.update(:transformation_mapping => transformation_mapping) }
```
[the remark this answers is not in the document]
Comment (lpichler) [reply app/controllers/api/transformation_mappings_controller.rb]: I used it 👍 (of course with @djberg96 approval :) )

**Comment by lpichler** [comment]
thanks @agrare @djberg96 👍

**Comment by djberg96** [comment]
The PR of the beast!

**Comment by simaishi** [comment]
@lpichler @djberg96 can this be `ivanchuk/yes`?

**Comment by djberg96** [comment]
@simaishi I think it's ok. @lpichler any objections?

**Comment by lpichler** [comment]
> @simaishi I think it's ok. @lpichler any objections?

No ,I don't have any.so can go to `ivanchuk`. Thanks @djberg96.

**Comment by simaishi** [comment]
Ivanchuk backport details:

```
$ git log -1
commit b233df92d1c1af00048280188166498489da75af
Author: Alberto Bellotti <abellotti@users.noreply.github.com>
Date:   Thu Aug 29 12:55:17 2019 -0400

    Merge pull request #666 from lpichler/use_factory_bot_for_transformation_mapping_item
    
    Create TransformationMapping before Item (FIX CI FAILURE)
    
    (cherry picked from commit 567572881a3d61093341cf581bec4d0e2bd2abcb)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1768517
```

**Pull request merged**